certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -167,6 +176,19 @@ def unsaved_files(self):  # pragma: no cover
         """Returns a list of unsaved filepaths"""
         return [assertions.PASS]
 
+    def _create_commentnode(self, path):
+        """Helper function to create a CommentNode from Augeas path"""
+
+        comment = self.parser.aug.get(path)
+        metadata = {"augeasparser": self.parser, "augeaspath": path}
+
+        # Because of the dynamic nature, and the fact that we're not populating
+        # the complete ParserNode tree, we use the search parent as ancestor
```

Comment:
wait, really? this will work? won't we want to be able to do things like modify comments, or delete them or something? or are we assuming this is essentially a get-only function that just needs to be able to find and read things?

Reply:
Oh, absolutely! We will have the ability to do so. All the mentioned actions will work directly using the `metadata["augeaspath"]`. This is also the reason we don't strictly need to prepopulate the tree, and can use the existing parser functionality for the plumbing.

Reply:
Note: Out of band, we decided we use `assertions.PASS` instead of the behavior described in the comment. On which note, we should probably remove this comment.

Reply:
I changed the comment to reflect the current state.